TGEvent's Metadata String Field Cause Deserialization Exception

If the metadata field actually includes a JSON object instead of a plain string, GSON cannot deserialize into a String. Should the field type be `JsonObject` or a `Map`? 

Exception:

```
01-06 10:42:16.635 27715-27715/com.femlite.app E/Tapglue: java.lang.IllegalStateException: Expected a string but was BEGIN_OBJECT at line 1 column 691 path $.events[1].metadata
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected a string but was BEGIN_OBJECT at line 1 column 691 path $.events[1].metadata
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:221)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:40)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:117)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:217)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.google.gson.Gson.fromJson(Gson.java:861)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.google.gson.Gson.fromJson(Gson.java:826)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at retrofit.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:36)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at retrofit.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:24)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at retrofit.OkHttpCall.parseResponse(OkHttpCall.java:148)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at retrofit.OkHttpCall.access$100(OkHttpCall.java:29)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at retrofit.OkHttpCall$1.onResponse(OkHttpCall.java:94)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.squareup.okhttp.Call$AsyncCall.execute(Call.java:177)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.squareup.okhttp.internal.NamedRunnable.run(NamedRunnable.java:33)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at java.lang.Thread.run(Thread.java:818)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err: Caused by: java.lang.IllegalStateException: Expected a string but was BEGIN_OBJECT at line 1 column 691 path $.events[1].metadata
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.google.gson.stream.JsonReader.nextString(JsonReader.java:838)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.google.gson.internal.bind.TypeAdapters$16.read(TypeAdapters.java:422)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.google.gson.internal.bind.TypeAdapters$16.read(TypeAdapters.java:410)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:117)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:217)
01-06 10:42:16.635 27715-27715/com.femlite.app W/System.err:    ... 17 more
```

--- a/tapglue-android-sdk/src/main/java/com/tapglue/model/TGEvent.java
+++ b/tapglue-android-sdk/src/main/java/com/tapglue/model/TGEvent.java
@@ -19,6 +19,7 @@ package com.tapglue.model;
 
 import android.support.annotation.NonNull;
 
+import com.google.gson.JsonElement;
 import com.google.gson.annotations.Expose;
 import com.google.gson.annotations.SerializedName;
 import com.tapglue.Tapglue;
@@ -40,7 +41,7 @@ public class TGEvent extends TGBaseObjectWithId<TGEvent> {
     private Float mLongitude;
     @Expose
     @SerializedName("metadata")
-    private String mMetadata;
+    private JsonElement mMetadata;
     @Expose
     @SerializedName("object")
     private TGEventObject mObject;
@@ -212,25 +213,25 @@ public class TGEvent extends TGBaseObjectWithId<TGEvent> {
     }
 
     /**
-     * Get metadata Metadata can contain more complex objects, all that is needed to support it is
-     * serialization and deserialization of them to json/xml/custom string format
+     * Get Metadata returns a JsonElement which can contain complex objects. The deserialization
+     * needs to be handled / implemented based on the custom format.
      *
      * @return metadata
      */
-    public String getMetadata() {
+    public JsonElement getMetadata() {
         return mMetadata;
     }
 
     /**
-     * Set metadata Metadata can contain more complex objects, all that is needed to support it is
-     * serialization and deserialization of them to json/xml/custom string format
+     * Set Metadata is a JsonElement and thus can contain more complex objects. The construction of
+     * the complex JsonElement or simply JsonPrimitive has to be handled / implemented.
      *
      * @param newValue new metadata value
      *
      * @return Current object
      */
     @NonNull
-    public TGEvent setMetadata(String newValue) {
+    public TGEvent setMetadata(JsonElement newValue) {
         mMetadata = newValue;
         return this;
     }

